### PR TITLE
[Examples] Correct run ner label2id for fine-tuned models

### LINKED ISSUE #14230: CoNLL2003 ner_tags order mismatch between the dataset from HF and the pretrained model
@patrickvonplaten @dslim23 

@dslim23 's pretrained models such as:
https://huggingface.co/dslim/bert-base-NER
have the following NER tag order baked in:
`"O", "B-MISC", "I-MISC", "B-PER", "I-PER", "B-ORG", "I-ORG", "B-LOC", "I-LOC"`

while the https://huggingface.co/datasets/conll2003 dataset has:
`O (0), B-PER (1), I-PER (2), B-ORG (3), I-ORG (4) B-LOC (5), I-LOC (6) B-MISC (7), I-MISC (8)`

The mismatch leads to defunct accuracy measurements out of the box for the pretrained NER models; try, for instance:
`python examples/pytorch/token-classification/run_ner.py --model_name_or_path dslim/bert-base-NER --dataset_name conll2003 --output_dir /tmp/test-ner --do_eval`

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Fixes #14230 . I'm really not sure about this PR however, so I'd like to wait for @sgugger's review here.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/pytorch/token-classification/run_ner.py [hunk 1]
```diff
@@ -296,31 +297,17 @@ def get_label_list(labels):
...
-        if label.startswith("B-") and label.replace("B-", "I-") in label_list:
-            b_to_i_label.append(label_list.index(label.replace("B-", "I-")))
-        else:
-            b_to_i_label.append(idx)
+    num_labels = len(label_list)
 
-    # Load pretrained model and tokenizer
-    #
-    # Distributed training:
-    # The .from_pretrained methods guarantee that only one local process can concurrently
```
Comment (sgugger) [inline examples/pytorch/token-classification/run_ner.py:317]: Why delete those comments?

File: examples/pytorch/token-classification/run_ner.py [hunk 2]
```diff
@@ -363,6 +350,31 @@ def get_label_list(labels):
             "requirement"
         )
 
+    if model.config.label2id != PretrainedConfig(num_labels=num_labels).label2id:
+        # Some have all caps in their config, some don't.
+        label_name_to_id = {k.lower(): v for k, v in model.config.label2id.items()}
```
Comment (sgugger) [inline examples/pytorch/token-classification/run_ner.py:355]: I think labels are all uppercased for NER so this line will be harmful (for text classification they are always lowercased coming from datasets).

**Review by sgugger — APPROVED** [review]
Thanks for working on this! Can you duplicate the same on `run_ner_no_trainer` as well?

**Comment by patrickvonplaten** [comment]
@sgugger, could you take a 2nd look here? :-)

File: examples/pytorch/token-classification/run_ner_no_trainer.py [hunk 3]
```diff
@@ -372,6 +365,31 @@ def get_label_list(labels):
 
     model.resize_token_embeddings(len(tokenizer))
 
+    if model.config.label2id != PretrainedConfig(num_labels=num_labels).label2id:
+        # Some have all caps in their config, some don't.
```
Comment (sgugger) [inline examples/pytorch/token-classification/run_ner_no_trainer.py:369]:
```suggestion
```

**Review by sgugger — APPROVED** [review]
LGTM! Thanks for fixing!
